batch/batch/front_end/front_end.py

```diff
@@ -1418,46 +1449,53 @@ async def delete(tx):
     return web.HTTPFound(deploy_config.external_url('batch', '/billing_projects'))
 
 
-@routes.post('/billing_projects/{billing_project}/users/add')
-@prom_async_time(REQUEST_TIME_POST_BILLING_PROJECT_ADD_USER_UI)
-@check_csrf_token
-@web_authenticated_developers_only(redirect=False)
-async def post_billing_projects_add_user(request, userdata):  # pylint: disable=unused-argument
-    db = request.app['db']
-    post = await request.post()
-    user = post['user']
-    billing_project = request.match_info['billing_project']
-
-    session = await aiohttp_session.get_session(request)
-
+async def _add_user_to_billing_project(db, billing_project, user):
     @transaction(db)
     async def insert(tx):
         row = await tx.execute_and_fetchone(
             '''
-SELECT billing_projects.name as billing_project, user
-FROM billing_projects
+SELECT billing_projects.name as billing_project,
+billing_projects.`status` as `status`,
+user FROM billing_projects
 LEFT JOIN (SELECT * FROM billing_project_users
-    WHERE billing_project = %s AND user = %s FOR UPDATE) AS t
-  ON billing_projects.name = t.billing_project
-WHERE billing_projects.name = %s;
-''',
+WHERE billing_project = %s AND user = %s FOR UPDATE) AS t
```

Comment:
Hmm. Not relevant to your change, but I think we ought to lock `billing_projects` as well. It'd be extra confusing if you're able to race to insert a user into a billing project with deleting that billing project.

Reply:
Yeah, I'll lock this.